patroni/ctl.py

```diff
@@ -556,17 +556,17 @@ def failover(config_file, cluster_name, master, candidate, force, dcs, scheduled
 
     config, dcs, cluster = ctl_load_config(cluster_name, config_file, dcs)
 
-    if cluster.leader is None and not is_paused(cluster):
+    if cluster.leader is None and not cluster.is_paused():
         raise PatroniCtlException('This cluster has no master')
 
-    if master is None and (not is_paused(cluster) or cluster.leader):
+    if master is None and (not cluster.is_paused() or cluster.leader):
         if force:
             master = cluster.leader.member.name
         else:
             master = click.prompt('Master', type=str, default=cluster.leader.member.name)
 
-        if not is_paused(cluster) and cluster.leader.member.name != master:
-            raise PatroniCtlException('Member {0} is not the leader of cluster {1}'.format(master, cluster_name))
+    if not (master is not None and cluster.leader and cluster.leader.member.name == master):
```

Comment:
Logic is broken, this statement can be rewritten as:
```python
if master is None or not cluster.leader or cluster.leader.member.name != master:
```

So exception will be raised for example if the master is None, what is totally wrong.